src/pipecat/services/openpipe.py

```diff
@@ -0,0 +1,159 @@
+from pipecat.services.ai_services import LLMService
+from openpipe import AsyncOpenAI as OpenPipeAI
+from openpipe import AsyncStream
+import os
+from loguru import logger
+import secrets
+import time
+import base64
+from openai.types.chat import (ChatCompletionMessageParam, ChatCompletionChunk)
+from typing import List
+from pipecat.processors.frame_processor import FrameDirection
+from pipecat.processors.aggregators.openai_llm_context import OpenAILLMContext, OpenAILLMContextFrame
+from pipecat.frames.frames import (
+    ErrorFrame,
+    Frame,
+    LLMFullResponseEndFrame,
+    LLMFullResponseStartFrame,
+    LLMMessagesFrame,
+    LLMResponseEndFrame,
+    LLMResponseStartFrame,
+    TextFrame,
+    URLImageRawFrame,
+    VisionImageRawFrame
+)
+
+
+class BaseOpenPipeLLMService(LLMService):
+
+    def __init__(
+            self,
+            model: str,
+            c_id=None,
+            api_key=None,
+            openpipe_api_key=None,
+            openpipe_base_url=None,
+            prompt=None):
+        super().__init__()
+        self._model = model
+        self._client = self.create_client(
+            api_key=api_key,
+            openpipe_api_key=openpipe_api_key,
+            openpipe_base_url=openpipe_base_url)
+        self.c_id = c_id if c_id else secrets.token_urlsafe(16)
+        self.prompt = prompt
+        logger.debug(f"Client Created: {self._client}")
+
+    def create_client(self, api_key=None, openpipe_api_key=None, openpipe_base_url=None):
```

Comment:
probably better to make this internal `def _create_client(...)` and remove default `None` values:

```
def _create_client(self, api_key: str | None, openpipe_api_key: str | None, openpipe_base_url: str | None):
```

Reply:
Sorry, forgot about this comment. I'll need to change something in `BaseOpenAILLMService`.